dist/flags.py

```diff
@@ -41,6 +41,12 @@ def flag_declare(name):
                         " GENERATION STOP 32", file=sys.stderr)
                     sys.exit(1)
                 end = int(m.group(0))
+
+                poweroftwo = (end != 0) and ((end & (end-1)) == 0)
+                if not poweroftwo and end != 12:
+                    print(name + ": line " + str(lcnt) + ", the stop value " + str(end) +
+                    " is not a power of 2", file=sys.stderr)
+                    sys.exit(1)
                 # Compare the number of flags defined and against the number
```

Comment:
I think you should consider adding in both script changes. If someone tried to make the original change, but changed it to 16 instead of 9 but left the field declaration at uint8_t, we'd want to catch it. Almost all uses of the flag generation have the flag field declared on the line after the STOP so grabbing the size and comparing it to the stop value would test most cases.

Reply:
Yeah, I will add both the script changes and test with different combinations in my local machine.